Modules/SubModule_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -244,15 +244,23 @@ function xASL_wrp_ResampleASL(x)
     elseif x.modules.asl.bTimeEncoded
         % Decoding of TimeEncoded data (Nifti is saved inside the function)
         ASL_im = xASL_quant_HadamardDecoding(PathASL{iSpace}, x.Q);
+        PLD_vector_size = length(unique(x.Q.Initial_PLD));
+        blockSize = x.Q.NumberEchoTimes * (PLD_vector_size - (PLD_vector_size/x.Q.TimeEncodedMatrixSize)); % Number of volumes per Hadamard block = EchoTimes*DecodedMatrixSize
         
-        % Hadamard Block size is calculated as number of TEs and the HadamardMatrixSize-1
-		blockSize = x.Q.NumberEchoTimes * (x.Q.TimeEncodedMatrixSize-1);
-		        
-        PWI = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), blockSize); % preallocate PWI
-        
-        for iBlock = 1:blockSize
-            PWI(:,:,:,iBlock) = xASL_stat_MeanNan(ASL_im(:,:,:,iBlock:blockSize:end), 4); % Averaged PWI4D across repetitions
+        if blockSize ~= size(ASL_im,4) % In case there is more than 1 block, we will average across repetitions

```

Comment:
Is "Size" here duration? A block is a single LabelingDuration right? Should this then be `blockDuration` to clarify this?

Reply:
OK then this should be blockCount or number right? Size suggests a multimensional count, for a 1D count (e.g., sum, count, number of) you don't usually use Size.

In fact, `blockSize` suggests a continuous measure, like the duration of a single labeling block. Not a count of integers, like the number of blocks.

Unless this is used by everyone and chosen in OSIPI TF4.1, we should avoid confusing language.

Reply:
I don't agree. We are not counting blocks, we are not labeling the blocks, we are saying how big a block is - how many fields it has. 

Sorry, but your comment does not really make sense. BlockSize is perfectly fitting for this. Same as vector size or vector length refer to the number of fields. Size doesn't suggest a continuous measure at all.

Reply:
You say vector length (1D) and matrix size (>1D) right?